Automattic/jetpack
# Payments: add new Payments post publish panel

#### Changes proposed in this Pull Request:

﻿This adds a new editor plugin, available in the following scenarios:

- When the site is connected to WordPress.com, or on WordPress.com Simple.
- When the site uses a plan that supports the Payments feature. On WordPress.com, this will require changes to add that new Payments feature to the list mapping plans and features.
- After a post has been published.

**Notes**

* This is a first version, that @niranjan-uma-shankar can expand on later on.
* I'm not sure about the name of the new feature. It's very generic, and I worry it could conflict with upcoming Payments things. Maybe we should rename it to `payments-post-publish`? @Copons What do you think?
* This is currently only displayed for folks who have a plan. I don't know if this should display a different message to folks without a plan?
* I have not fully tested this, and more specifically, I have not tested this on pages; I believe the intent is to only show this on pages, not posts; I have not implemented this.

**Screenshot**

<img width="310" alt="image" src="https://user-images.githubusercontent.com/426388/155736944-4711ee2b-9bff-4822-9936-e0993a547be1.png">


#### Jetpack product discussion

* PT: pcbrnV-3Op-p2
* Slack: p1645778718984779-slack-CJS75TX3R

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Connect your site to WordPress.com.
* Purchase a bundled plan (like Complete)
* Go to Pages > Add New
* Write and Publish a new page.
* Notice the new plugin in the post publish panel.
* Click on the link.
* Check Tracks Live View for `jetpack_editor_payments_post_publish_click` 5 minutes later; you should see your click.
* Verify that the post publish panel _does not_ appear when publishing a new post via Posts > Add New.
